src/services/ls-migration/addressBook.ts

```diff
@@ -26,3 +27,45 @@ export const migrateAddressBook = (lsData: LOCAL_STORAGE_DATA): AddressBookState
     }
   }
 }
+
+// Temporary post-migration fix for malformed data
+export const sanitizeMigratedAddressBook = (state: AddressBookState): AddressBookState => {
+  const hasValidAb = Object.values(state).every((ab) => {
+    return Object.keys(ab).every(utils.isAddress)
+  })
+
+  if (hasValidAb) {
+    return state
+  }
+
+  const sanitizedAb = Object.keys(state).reduce<AddressBookState>((sanitizedAb, chainId) => {
+    const chainAb = state[chainId]
+
+    const sanitizedChainAb = Object.keys(chainAb).reduce<AddressBook>((sanitizedChainAb, address) => {
+      if (!address || !utils.isAddress(address)) {
```

Comment:
We checksum addresses addresses on entry in both `safe-react` and `web-core`. Whilst it would be nice to check them again, I don't think we should modify the values.

I've added logs to inform users if an address was removed.